source/include/FreeRTOS_IP_Private.h

```diff
@@ -606,7 +606,7 @@ BaseType_t xIPIsNetworkTaskReady( void );
         uint8_t ucRepCount;            /**< Send repeat count, for retransmissions
                                         * This counter is separate from the xmitCount in the
                                         * TCP win segments */
-        uint8_t ucTCPState;            /**< TCP state: see eTCP_STATE */
+        eIPTCPState_t ucTCPState;      /**< TCP state: see eTCP_STATE */
```

Comment:
Initially I never used enum's as a struct member because their size is unpredictable, it may vary from 8 to 32-bytes.
The advantage of storing it as a char is that the we can control the alignment of the struct.

I know that MISRA doesn't like the conversions from `uint8_t` to an `enum` and back because theoretically they might lead to a loss of bits. We know that the conversions are safe by design ( there are only 12 TCP states ).

Only one remark: variables and fields of type `enum` start with the prefix `e`, so it should be `eIPTCPState_t eTCPState`.

Reply:
I didn't change it initially for this reason, but I got a comment to change it.
What do you suggest in that case? revert back to a fixed size or keep it as it is? 
we could also put the states as defines.
alternatively we could use 
```
enum __attribute__ ((__packed__)) my_enum {...};
```
this will make the enum 1 byte
we already have an include "packed.h"  ready

Reply:
> I didn't change it initially for this reason, but I got a comment to change it. What do you suggest in that case? revert back to a fixed size or keep it as it is? we could also put the states as defines. alternatively we could use
> 
> ```
> enum __attribute__ ((__packed__)) my_enum {...};
> ```
> 
> this will make the enum 1 byte we already have an include "packed.h" ready, with a MISRA exception for it

Haha, I haven't seen tha packed attribute on an enum declaration. That sounds like an ideal solution.

Reply:
@htibosch wrote:
> Initially I never used enum's as a struct member because their size is unpredictable

Yes, I thought of that but could not come up with a good reason of not leaving it up to the compiler. And then you proposed:

> The advantage of storing it as a char is that the we can control the alignment of the struct

I did think of this as well. But, is there a very strong reason that we want these `structs` to be aligned?